pyro/contrib/epidemiology/compartmental.py

```diff
@@ -355,6 +352,13 @@ def heuristic():
         mcmc = MCMC(kernel, **options)
         mcmc.run()
         self.samples = mcmc.get_samples()
+        if haar:
+            # Transform back from Haar coordinates.
+            x = self.samples.pop("auxiliary_haar")
+            x = HaarTransform(dim=-2 if self.is_regional else -1, flip=True).inv(x)
+            x = biject_to(constraints.interval(-0.5, self.population + 0.5))(x)
+            self.samples["auxiliary"] = x
+
```

Comment:
Previously we applied the reparam during `.predict()`, but it is simpler (and fixed weird bugs) to instead transform directly in `.fit()`. This has the added benefit of isolating reparam logic within the `.fit()` method: `auxiliary_haar` is immediately replaced by `auxiliary`.